src/commands/StepCommand.java

```diff
@@ -28,12 +33,41 @@ public void execute(String[] input, SimulationManager simulationManager) {
         }
 
         int steps = 1;
+        List<String> constraints = new ArrayList<String>();
+
         if (input.length > 1) {
```

Comment:
We currently are not correctly handling cases when the user includes both n and constraints. I think it should just be a simple error message.
```
(aldb) step 2 ["", "Fox in far"]

S3
----
far: {  }
near: { Farmer, Fox, Grain }
```

Reply:
We don't print an error if the user enters `step 2 2` either.

Consequently, I think this is best addressed in a separate change (should we choose to enforce strict parameter limits on all our commands).